.cm/gitstream.cm

```diff
@@ -0,0 +1,81 @@
+# -*- mode: yaml -*-
+manifest:
+  version: 1.0
+
+automations:
+  # Enforces a certain format on PR titles
+  enforce_pr_title:
+    if:
+      - {{ pr.title | match(regex=titlePolicy.titleRegex) | nope }}
+    run:
+      - action: request-changes@v1
+        args:
+          comment: |
+            All PRs must be titled according to our semantic naming policy: `[Area]: <short summary>`
+
+            Area must be one of the following:
+             * DataGrid
+             * Pickers 
+             * Charts
+             * TreeView
+             * Chore
+  # Add a label that indicates how many minutes it will take to review the PR.
+  estimated_time_to_review:
+    if:
+      - true
+    run:
+      - action: add-label@v1
+        args:
+          label: "{{ calc.etr }} min review"
+          color: {{ colors.error if (calc.etr >= 20) else (colors.warning if (calc.etr >= 5) else colors.success) }}
+  # Post a comment that lists the best experts for the files that were modified.
+  explain_code_experts:
+    if:
+      - { { pr.labels | match(term='Auto assign') | some } }
+    run:
+      - action: explain-code-experts@v1
+        args:
+          gt: 10
+  # Assign a random reviewer when the 'Share knowledge' label gets added
+  share_knowledge:
+    if:
+      - { { pr.labels | match(term='Share Knowledge') | some } }
+    run:
+      - action: add-reviewers@v1
+        args:
+          reviewers: { { repo | codeExperts(gt=30, lt=60) | random } }
+      - action: add-comment@v1
+        args:
+          comment: |
+            gitStream has assigned a reviewer to increase knowledge sharing on this PR.
```

Comment:
I'm very curious to see how gitStream can assign a relevant reviewer depending on the modified files.

Reply:
If we can select the pool of people to chose at random for that would be ideal as people might switch teams but still be invited to review PRs from components they no longer maintain.

Reply:
thats an interesting approach ... I probably need to dig the docs about this as i have not seen a mention of restricting the pool to choose from. Maybe I'll contact the devs about that

Reply:
I think that PR reviewers assignment should have the very first criterion as the GitHub label assigned, following  https://www.notion.so/mui-org/GitHub-community-issues-PRs-Tier-1-12a84fdf50e44595afc55343dac00fca?pvs=4#c6b06804e0ac40c3aa2b5b5c16b202bf, then it's about finding the list of maintainers, and lastly, about figuring out among this pool, the most relevant reviewers, balancing people who already have the context, with the need to increase codebase awareness for the other maintainers.

Reply:
It sounds like step one is identifying the correct reviewers for PRs, so I recommend checking out the options we highlight on [review assignment](https://docs.gitstream.cm/automations/standard/review-assignment). In particular, you can create watchlists to notify teams or individuals based on modified resources, assign someone from the author's team, or flag specific locations for additional required reviews.

From there, you can create additional automations to apply labels based on the people who are assigned to the PR.

Reply:
The maintainers list per GitHub labels is available through a Notion database API call.